ocp_diagnosis.sh

```diff
@@ -123,8 +134,47 @@ if [[ "$PROMETHEUS_CAPTURE" == "true" ]]; then
 		echo "Looks like $type is not a valid option, please check"
 		help
 	fi
+}
+
+
+function set_pbench() {
+	echo "Detected storage mode as $STORAGE_MODE"
+	echo "Assuming that the ocp diagnosis tool is run using pbench-user-benchmark"
+	echo "Fetching the latest pbench results dir"
+	result_dir="/var/lib/pbench-agent/$(ls -t /var/lib/pbench-agent/ | grep "pbench-user" | head -1)"/1-default/reference-result
+	OUTPUT_DIR="/var/lib/pbench-agent/$(ls -t /var/lib/pbench-agent/ | grep "pbench-user" | head -1)"/1-default
+	echo "Copying the collected data to $result_dir"
+}
+
+
+function store() {
+	# parameters
+	# 	1 function to capture data
+	# 	2 filename
+
+	if [[ -z $STORAGE_MODE ]]; then
+		# store it locally, i.e on the jump host itself
+		echo "Looks like STORAGE_MODE is not defined, storing the results on local file system"
+		$1;
+	elif [[ $STORAGE_MODE == "pbench" ]]; then
+		# store on the pbench server (STORAGE_MODE=pbench)
+		set_pbench;
+		$1;
+	elif [[ validate_server_is_up -eq 0 ]]; then
+		# store it on the data server (STORAGE_MODE=http://ec2.0.0.0.0:7070)
+		$1 && validate_server_is_up && post_to_server "$OUTPUT_DIR/$2"
+	else
+		echo "Invalid storage mode chosen. STORAGE_MODE is $STORAGE_MODE"
```

Comment:
Maybe exit 1 if the storage mode set is not valid?

Reply:
added